packages/manager/src/features/CloudPulse/Utils/FilterBuilder.ts

```diff
@@ -743,3 +745,24 @@ export const filterEndpointsUsingRegion = (
 
   return data.filter(({ region }) => region === regionFromFilter);
 };
+
+/**
+ * @param serviceType The service type for which the filter needs to be applied
+ * @param data The resources for which the filter needs to be applied
+ * @returns The filtered resources
+ */
+export const filterUsingSpecialConditions = (
```

Comment:
```suggestion
export const filterFirewallResources = (
```

lets give a proper name

Reply:
This is a generic function